### PR TITLE
Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by rhiroy**
Adds a section to the Contributing.md documentation to include a list of useful terminology and resources to help orient new contributors to the technologies relevant to Styled Material Components.

### CODE REVIEW

**Comment by brad-decker** [comment]
There's nothing wrong with mentioning material UI -- That library is a very well known and respected one and just because we offer a different path doesn't make us competitors. In many ways the inspiration for this package comes from learnings had by utilizing theirs.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -53,3 +54,25 @@ Wherever possible, do not take these conversations to private channels, includin
 Follow us on [Twitter](https://twitter.com/merlinOSA) to learn more about our open source initiatives and get involved with our community <3
 
 Read more about contributing to the open source community on Github's fantastic [Open Source Guides](https://opensource.guide/)
+
+## Resources and Terminology
+
+* **[React](https://reactjs.org/)**
+is a component-based library for for managing state and creating user interfaces.
+* **[styled-components](https://www.styled-components.com/)**
+Inspired by the component-based architecture of React, Max Stoiber and Glenn Maddern created a library, styled-components that allows you to style your application using React components.
```
Comment (martahj) [inline CONTRIBUTING.md]: Tiny grammar nit, but I think there should be a comma after "styled-components":
```
Inspired by the component-based architecture of React, Max Stoiber and Glenn Maddern created a library, styled-components, that allows you to style your application using React components.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -8,6 +8,7 @@ Welcome to Styled Material Components! We're so glad you're here. Please read th
 0. [How to contribute](#how-to-contribute)
 0. [Contribution review process](#contribution-review-process)
 0. [Community](#community)
+0. [Resources and Terminology]
```
Comment (AriLFrankel) [inline CONTRIBUTING.md]: Was the intention to link this to the Resources and Terminology section below?

Comment (AriLFrankel) [reply CONTRIBUTING.md]: @rhiroy

**Pull request merged**